sycl/test-e2e/PropagateOptionsToBackend/sycl-opt-level.cpp

```diff
@@ -0,0 +1,50 @@
+//==----------- sycl-opt-level.cpp  - DPC++ SYCL on-device test
+//---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This test verifies the propagation of front-end compiler optimization
+// option to the backend.
+// API call in device code:
+// Following is expected addtion of options:
+// Front-end option | OpenCL backend option | L0 backend option
+//       -O0        |    -cl-opt-disable    |  -ze-opt-disable
+//       -O1        |    /* no option */    |  -ze-opt-level=1
+//       -O2        |    /* no option */    |  -ze-opt-level=1
+//       -O3        |    /* no option */    |  -ze-opt-level=2
+
+// RUN: %clangxx -O0 -fsycl %s -o %t0.out
+// RUN: env ONEAPI_DEVICE_SELECTOR=level_zero:gpu SYCL_PI_TRACE=-1 %t0.out 2>&1 %GPU_CHECK_PLACEHOLDER --check-prefixes=CHECK0
```

Comment:
I don't think it's good. What if I'm running the test suite on a system with no L0 installed (only `opencl:gpu`). This test overrides the harness' device selection...

Reply:
I split the test into two - one for opencl and one for level_zero with '// REQUIRES: opencl and //REQUIRES: level_zero respectively. Thanks